mne/channels/channels.py

```diff
@@ -440,18 +440,36 @@ def inst_has(attr):
 class InterpolationMixin(object):
     """Mixin class for Raw, Evoked, Epochs
     """
-    def interpolate_bads_eeg(self):
-        """Interpolate bad channels
 
-        Operates in place.
+    def interpolate_bads(self, reset_bads=True):
```

Comment:
On Tue, Apr 28, 2015 at 3:55 PM, Alexandre Gramfort <
notifications@github.com> wrote:

> In mne/channels/channels.py
> <https://github.com/mne-tools/mne-python/pull/2011#discussion_r29244217>:
>
> >
> > -        Operates in place.
> > +    def interpolate_bads(self, reset_bads=True):
>
> Writing evoked_interp = evoked.interpolate_bads() When evoked is modified
> inplace is what is wrong.
>

but isn't this what is done in most epoch methods?


> If you return nothing and use copy() method when needed I am happy
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/mne-tools/mne-python/pull/2011/files#r29244217>.
>